.github/workflows/ci.yml

```diff
@@ -79,3 +79,26 @@ jobs:
     steps:
       - name: Check whether all jobs pass
         run: echo '${{ toJson(needs) }}' | jq -e 'all(.result == "success")'
+
+  # Publish snapshot packages. These jobs will only be triggered when the CI is executed on main.
+  publish_jvm_snapshot_package:
+    name: Publish JVM snapshot package
+    if: github.ref == 'refs/heads/main'
+    needs: ci
+    uses: ./.github/workflows/publish_jvm.yml
+    permissions:
+      contents: read
+      packages: write
+    with:
+      version: 0.11.0-rc-SNAPSHOT
```

Comment:
While this is technically valid [semver](https://semver.org/), it is quite confusing (i.e. being a snapshot of a release candidate). I suggest going with `0.11.0-SNAPSHOT` in this case, as it means roughly the same thing as `0.11.0-dev`.

Reply:
Starting from the next release, we will drop the `-rc` pre-release identifier so we should be doing `major.minor.patch-SNAPSHOT` anyway.